// many-pictures/es/src/pic-viewer.vue_vue_type_script_setup_true_lang.mjs
import { defineComponent as u, useCssVars as r, unref as c, computed as p, withDirectives as m, openBlock as f, createElementBlock as _, createElementVNode as n, normalizeStyle as g, vShow as v, pushScopeId as I, popScopeId as h } from "vue";
const k = (e) => (I("data-v-6130363f"), e = e(), h(), e), x = { class: "pic-viewer" }, z = /* @__PURE__ */ k(() => /* @__PURE__ */ n("svg", { id: "svg-blur" }, [
  /* @__PURE__ */ n("filter", { id: "blur" }, [
    /* @__PURE__ */ n("feGaussianBlur", { stdDeviation: "100" })
  ])
], -1)), V = ["src"], w = /* @__PURE__ */ u({
  name: "pic-viewer",
  props: {
    imgLink: null,
    modelValue: { type: Boolean },
    zIndex: null
  },
  emits: ["close", "open", "update:modelValue"],
  setup(e, { emit: l }) {
    const o = e;
    r((s) => ({
      "26e195fc": c(i),
      "101439b6": d
    }));
    const i = o.zIndex, d = o.zIndex + 1, t = p({
      get: () => o.modelValue,
      set: (s) => {
        l("update:modelValue", s), l(s ? "open" : "close");
      }
    });
    function a() {
      t.value = !1;
    }
    return (s, b) => m((f(), _("div", x, [
      z,
      n("div", {
        id: "background",
        style: g({ backgroundImage: `url(${o.imgLink})` }),
        onClick: a
      }, null, 4),
      n("img", {
        src: o.imgLink,
        id: "main-image"
      }, null, 8, V)
    ], 512)), [
      [v, c(t)]
    ]);
  }
});
export {
  w as default
};
